### ISSUE TITLE
Feature request: verify local data on completion of file download

### ISSUE THREAD

**Issue by CyberSkull** [transmission/transmission#4724]
### What is the issue?

Every now and then I get a download that isn't actually finished or is corrupted in Transmission. It would be good if when a file is downloaded, the full verify local data integrity check is run.

I've run into this problem on multiple builds and version of macOS.

### Which application of Transmission?

macOS app

**Comment by ckerr** [comment]
This sounds like you're discussing #4621, which is fixed by https://github.com/transmission/transmission/pull/4665.

There _is_ a PR at https://github.com/transmission/transmission/pull/4178 to do exactly what this issue requests, but it's undecided whether this is still useful after 4665.

**Comment by ckerr** [comment]
So I'm closing this as a duplicate but since there are two different tracks here I'm not sure which one it's a duplicate of :rofl:

**Comment by Lakis01** [comment]
I am running version 4.0.6 on Ubuntu, installed through snap and I am having the issue that many torrents are shown as completed , but the downloaded file is actually corrupted and the verify local data reveals that only about 95% of data is actually correct (for a specific torrent I have just checked). It hadn't occurred to me that this is possible and I had assumed that the files were corrupt on the uploaders / seeders end.

I often resume downloads after the computer has been suspended / shut down and was also running low on storage at some point (low free storage has been mentioned as a possible reason in one of the issues).

**Comment by valcomm** [comment]
I don't think this is a duplicate.  To me this one seems as a request to verify the data on event 'completed'.  As a setting or a hook

**Comment by RichyT** [comment]
> There _is_ a PR at [#4178](https://github.com/transmission/transmission/pull/4178) to do exactly what this issue requests, but it's undecided whether this is still useful after 4665.

It is. You don't know what happens after the checksum has been verified and the attempt to write is made.

This shouldn't even be controversial. The code to do the verification is already there and code to do something once the download is complete. You can even leave it disabled by default.
